src/SixLabors.Fonts/TextLayout.cs

```diff
@@ -85,18 +85,27 @@ public IReadOnlyList<GlyphLayout> GenerateLayout(ReadOnlySpan<char> text, Render
                     previousGlyph = null;
                 }
 
-                if (spanStyle.Font.LineHeight > unscaledLineHeight)
+                bool hasFourBytes = char.IsHighSurrogate(text[i]);
+                int codePoint = hasFourBytes ? char.ConvertToUtf32(text[i], text[i + 1]) : text[i];
+
+                GlyphInstance glyph = spanStyle.GetGlyph(codePoint);
+                if (glyph == null)
+                {
+                    throw new Exceptions.FontException($"Cannot find a glyph for code point {codePoint}");
```

Comment:
We should use a throwhelper in a loop. This might be inlinable.

Reply:
sorted